patina_dxe_core/src/gcd.rs

```diff
@@ -227,24 +304,31 @@ pub fn add_hob_resource_descriptors_to_gcd(hob_list: &HobList) {
                     )
                     .expect("Failed to add memory space to GCD");
                 }
-                if let Some(attributes) = memory_attributes {
-                    match GCD.set_memory_space_attributes(
-                        split_range.start as usize,
-                        split_range.end.saturating_sub(split_range.start) as usize,
-                        attributes,
-                    ) {
-                        // NotReady is expected result here since page table is not yet initialized. In this case GCD
-                        // will be updated with the appropriate attributes which will then be sync'd to page table
-                        // once it is initialized.
-                        Err(EfiError::NotReady) => (),
-                        _ => {
-                            panic!(
-                                "GCD failed to set memory attributes {:#X} for base: {:#X}, length: {:#X}",
-                                attributes,
-                                split_range.start,
-                                split_range.end.saturating_sub(split_range.start)
-                            );
-                        }
+                match GCD.set_memory_space_attributes(
+                    split_range.start as usize,
+                    split_range.end.saturating_sub(split_range.start) as usize,
+                    memory_attributes,
+                ) {
+                    // NotReady is expected result here since page table is not yet initialized. In this case GCD
+                    // will be updated with the appropriate attributes which will then be sync'd to page table
+                    // once it is initialized.
+                    Err(EfiError::NotReady) => (),
+                    Ok(()) => {
```

Comment:
Did this case actually arise? I think we should always get EfiError::NotReady here.